src/pages/Layout/ProfileQuestionLayout/QuestionList/index.tsx

```diff
@@ -1,26 +1,148 @@
+import { Loading } from '@/components/Loading'
+import { AlertModal } from '@/components/Modal/AlertModal'
+import { ConfirmModal } from '@/components/Modal/ConfirmModal'
+import ErrorPage from '@/pages/ErrorPage'
+import { useSelectedQuestionStore } from '@/stores/selected-question'
 
-interface QuestionListProps {
-  questions: QuestionItem[]
+type QuestionListProps = {
+  isMyPage: boolean
 }
 
-export const QuestionList = ({ questions }: QuestionListProps) => {
-  const questionId = useSeletedQuestionStore((state) => state.questionId)
-  const setSeletedQuestion = useSeletedQuestionStore(
+export const QuestionList = ({ isMyPage }: QuestionListProps) => {
+  const [selectedQuestionId, setSelectedQuestionId] = useState<number | null>(
+    null
+  )
+  const [contextMenuPosition, setContextMenuPosition] = useState<{
+    x: number
+    y: number
+  } | null>(null)
+  const contextMenuRef = useRef<HTMLDivElement | null>(null)
+  const deleteAlert = useDisclosure()
+  const errorAlert = useDisclosure()
+
+  const questionId = useSelectedQuestionStore((state) => state.questionId)
+  const setSelectedQuestion = useSelectedQuestionStore(
     (state) => state.setQuestionId
   )
+  const setQuestionContent = useSelectedQuestionStore(
+    (state) => state.setQuestionContent
+  )
+  const setQuestionCreatedAt = useSelectedQuestionStore(
+    (state) => state.setQuestionCreatedAt
+  )
+
+  const location = useLocation()
+  const userId: string = location.state?.userId.toString()
+
+  const {
+    data: questions,
+    isLoading,
+    error,
+  } = useGetProfileQuestion(userId || '')
+
+  const isFirstRender = useRef(true)
+
+  useEffect(() => {
+    const handleClickOutside = (e: MouseEvent) => {
+      if (
+        contextMenuRef.current &&
+        !contextMenuRef.current.contains(e.target as Node)
+      ) {
+        setContextMenuPosition(null)
+      }
+    }
+    document.addEventListener('mousedown', handleClickOutside)
+    return () => document.removeEventListener('mousedown', handleClickOutside)
+  }, [])
 
   useEffect(() => {
-    if (questions.length > 0) {
-      setSeletedQuestion(questions[0].profileQuestionId)
+    // 첫 렌더링 시 초기 store 지정
+    if (isFirstRender.current && questions && questions.length > 0) {
+      setSelectedQuestion(questions[0].profileQuestionId)
+      setQuestionContent(questions[0].profileQuestionContent)
+      setQuestionCreatedAt(questions[0].createdAt)
+
+      isFirstRender.current = false
+    } else if (isFirstRender.current && questions?.length === 0) {
+      setSelectedQuestion(undefined)
+      setQuestionContent(undefined)
+      setQuestionCreatedAt(undefined)
+    }
+  }, [
+    questions,
+    questionId,
+    setSelectedQuestion,
+    setQuestionContent,
+    setQuestionCreatedAt,
+  ])
+
+  const { mutate: deleteQuestion } = useMutation<
+    void,
+    Error,
+    DeleteProfileQuestionRequest
+  >({
+    mutationFn: ({ deleteQuestionId }: DeleteProfileQuestionRequest) =>
```

Comment:
```suggestion
    mutationFn: ({ deleteQuestionId }: DeleteProfileQuestionRequest) =>
```

타입추론이 자동으로 되어서 타입을 지정안해줘도 됩니당